packages/core/src/logout/endSessionUrl.ts

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+export interface IEndSessionOptions {
+  endSessionEndpoint: string;
+  idTokenHint?: string;
+  postLogoutRedirectUri?: string;
+  state?: string;
+}
```

Comment:
I think you could do something like 
```
export interface IEndSessionOptions {
  endSessionEndpoint: string;
  idTokenHint?: string;
  postLogoutRedirectUri?: string;
  state?: string;
}

export interface IEndSessionOptionsPostLogout extends IEndSessionOptions {
  postLogoutRedirectUri: string;
  state?: string;
}

export interface IEndSessionOptionsNoCallback extends IEndSessionOptions {
  postLogoutRedirectUri?: undefined;
  state?: undefined;
}
```
which would result in a compile-time error if "state" is specified but not `postLogoutRedirectUri`:
```
const test = (a: IEndSessionOptionsPostLogout | IEndSessionOptionsNoCallback) : void => {};
test({endSessionEndpoint: "test", state: "test"})
```
The names of the interfaces here are a bit too long, but they show the intent. I would typically do this inline with a union type, but I don't think interfaces allow such thing. Approaching it the same way as `ILogoutOptions` sounds fine to me.

Reply:
I find that these kinds of patterns are generally more of a hindrance than a help if consumers because the TS type inference system just doesn't work well with a lot of these types. To provide a minimal example as to why; the following code

```ts
interface T {
  a: true;
  b: true;
}

interface F {
  a: false;
  b: false;
}

function test1({ a, b }: T | F) {
  return test2({ a, b })
}

function test2(options: T | F) {
  throw new Error("Not Implemented")
}
```

 results in a typescript compilation error

```ts
Argument of type '{ a: boolean; b: boolean; }' is not assignable to parameter of type 'T | F'.
  Type '{ a: boolean; b: boolean; }' is not assignable to type 'F'.
    Types of property 'a' are incompatible.
      Type 'boolean' is not assignable to type 'false'.ts(2345)
```

Reply:
In fact we get exactly this kind of error internally if we do try and implement such changes.
 - Commit implementing these changes: https://github.com/inrupt/solid-client-authn-js/commit/1c8cec3c9cbe4143cdb7e0b750222081e19cafc1
 - Resulting TS Error: https://github.com/inrupt/solid-client-authn-js/actions/runs/5459532806/jobs/9935630385

Reply:
I've added some jsdoc in https://github.com/inrupt/solid-client-authn-js/pull/3027/commits/c72ee18497a41277af3c5cd75621b73fe268379f to try and address this.

Reply:
Well that's annoying. The following actually compiles for 1c8cec3, which I find ridiculous ^^
```
if (options.postLogoutUrl === undefined) {
        this.boundLogout({
          logoutType: "idp",
          postLogoutUrl: options.postLogoutUrl,
          state: options.state,
        });
      } else {
        this.boundLogout({
          logoutType: "idp",
          postLogoutUrl: options.postLogoutUrl,
          state: options.state,
        });
      }
```
Adding the docs is a good alternative solution.

Reply:
> The following actually compiles for https://github.com/inrupt/solid-client-authn-js/commit/1c8cec3c9cbe4143cdb7e0b750222081e19cafc1, which I find ridiculous ^^

This is something to take up with the TS team one day ...